docs/frontend.md

```diff
@@ -3,313 +3,186 @@ id: frontend
-        }
-      ]
-    }
-    case OPTIMISTIC_REMOVE_SHOPPING_LIST: {
-      return state.filter(item => {
-        return item.id !== action.payload.id
-      })
-    }
-    case OPTIMISTIC_SYNC: {
-      return action.payload.originalLists
-    }
-    default: {
-      return state
-    }
-  }
-}
+A connected component receives additional props. These props provide access to the Read Model data and Redux action creators mapped to reSolve commands.
 
-export default optimistic_shopping_lists
-```
+#### Example
+
+The [shopping-list](https://github.com/reimagined/resolve/tree/master/examples/shopping-list) example application demonstrates how to use the **resolve-client** library to implement a react-redux frontend for a reSolve application.
 
 <!-- prettier-ignore-end -->
 
-3. Implement an optimistic middleware to intercept actions used to communicate with a Read Model and update the Redux state accordingly:
-
-<!-- prettier-ignore-start -->
-
-[embedmd]:# (..\..\examples\shopping-list\client\middlewares\optimistic_shopping_lists_middleware.js /^/ /\n$/)
-
-```js
-import { actionTypes } from 'resolve-redux'
-
-import {
-  OPTIMISTIC_CREATE_SHOPPING_LIST,
-  OPTIMISTIC_REMOVE_SHOPPING_LIST,
-  OPTIMISTIC_SYNC
-} from '../actions/optimistic_actions'
-
-const { SEND_COMMAND_SUCCESS, LOAD_READMODEL_STATE_SUCCESS } = actionTypes
-
-const optimistic_shopping_lists_middleware = store => next => action => {
-  if (
-    action.type === SEND_COMMAND_SUCCESS &&
-    action.commandType === 'createShoppingList'
-  ) {
-    store.dispatch({
-      type: OPTIMISTIC_CREATE_SHOPPING_LIST,
-      payload: {
-        id: action.aggregateId,
-        name: action.payload.name
-      }
-    })
-  }
-  if (
-    action.type === SEND_COMMAND_SUCCESS &&
-    action.commandType === 'removeShoppingList'
-  ) {
-    store.dispatch({
-      type: OPTIMISTIC_REMOVE_SHOPPING_LIST,
-      payload: {
-        id: action.aggregateId
-      }
-    })
-  }
-  if (action.type === LOAD_READMODEL_STATE_SUCCESS) {
-    store.dispatch({
-      type: OPTIMISTIC_SYNC,
-      payload: {
-        originalLists: action.result
-      }
-    })
-  }
-
-  next(action)
-}
+## resolve-react-hooks library
 
-export default optimistic_shopping_lists_middleware
-```
+The **resolve-react-hooks** library provides React hooks that you can use to connect React components to a reSolve backend. The following hooks are provided.
 
-<!-- prettier-ignore-end -->
+| Hook                                                    | Description                                                             |
+| ------------------------------------------------------- | ----------------------------------------------------------------------- |
+| [useCommand](api-reference.md#usecommand)               | Initializes a command that can be passed to the backend                 |
+| [useCommandBuilder](api-reference.md#usecommandbuilder) | Allows to generate commands based on input parameters                   |
```

Comment:
Who does it allow to generate commands? Is it "you" (the reader/developer)?

Reply:
Changed to "Allows a component to"